internal/cli/plugin/plugin_asset.go

```diff
@@ -0,0 +1,253 @@
+	repositoryOwner  string
+	repositoryName   string
+	releaseVersion   string
+	pluginAssets     []*github.ReleaseAsset
+}
+
+func (opts *AssetOpts) fullRepositoryDefinition() string {
+	return fmt.Sprintf("%s/%s", opts.repositoryOwner, opts.repositoryName)
+}
+
+func (opts *AssetOpts) getPluginAssets() ([]*github.ReleaseAsset, error) {
+	var err error
+	var release *github.RepositoryRelease
+
+	// download latest release if version is not specified
+	if opts.releaseVersion == "" {
+		release, _, err = opts.ghClient.Repositories.GetLatestRelease(context.Background(), opts.repositoryOwner, opts.repositoryName)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not find latest release for %s", opts.fullRepositoryDefinition())
+		}
+	} else {
+		// try to find the release with the version tag with v prefix, if it does not exist try again without the prefix
+		release, _, err = opts.ghClient.Repositories.GetReleaseByTag(context.Background(), opts.repositoryOwner, opts.repositoryName, "v"+opts.releaseVersion)
+
+		if release == nil || err != nil {
+			release, _, err = opts.ghClient.Repositories.GetReleaseByTag(context.Background(), opts.repositoryOwner, opts.repositoryName, opts.releaseVersion)
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("could not find the release %s release for %s", opts.releaseVersion, opts.fullRepositoryDefinition())
+		}
+	}
+
+	return release.Assets, nil
+}
+
+func (opts *AssetOpts) getAssetID() (int64, error) {
+	operatingSystem, architecture := runtime.GOOS, runtime.GOARCH
+	for _, asset := range opts.pluginAssets {
+		if *asset.ContentType != "application/gzip" {
+			continue
+		}
+		name := *asset.Name
+
+		if strings.Contains(name, operatingSystem) && strings.Contains(name, architecture) {
+			return *asset.ID, nil
+		}
+	}
+
+	return 0, fmt.Errorf("could not find an asset to download from %s for %s %s", opts.fullRepositoryDefinition(), operatingSystem, architecture)
+}
+
+func (opts *AssetOpts) getPluginAssetAsReadCloser() (io.ReadCloser, error) {
+	assetID, err := opts.getAssetID()
+
+	if err != nil {
+		return nil, err
+	}
+
+	rc, _, err := opts.ghClient.Repositories.DownloadReleaseAsset(context.Background(), opts.repositoryOwner, opts.repositoryName, assetID, http.DefaultClient)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not download asset with ID %d from %s", assetID, opts.fullRepositoryDefinition())
+	}
+
+	return rc, nil
+}
+
+func parseGithubValues(arg string) (string, string, error) {
+	arg = strings.Split(arg, "@")[0]
+	arg = strings.TrimSuffix(arg, "/")
+
+	parts := strings.Split(arg, "/")
+
+	err := errors.New(`github parameter is invalid. It needs to have the format "<github-owner>/<github-repository-name>"`)
+
+	minParts := 2
+	if len(parts) < minParts {
+		return "", "", err
+	}
+	owner := parts[len(parts)-2]
+	name := parts[len(parts)-1]
+
+	if owner == "" || name == "" {
+		return "", "", err
+	}
+
+	return owner, name, nil
+}
+
+func parseReleaseVersion(arg string) (string, error) {
```

Comment:
Why not return a version here instead of a string?
You return an empty string for latest anyway.

Reply:
done